roboteam_ai/src/ApplicationManager.cpp

```diff
@@ -128,6 +128,8 @@ void ApplicationManager::runOneLoopCycle() {
         std::cout <<"NO FIRST WORLD" << std::endl;
         ros::Duration(0.2).sleep();
     }
+
+    weHaveRobots = ai::world::world->weHaveRobots();
```

Comment:
why?

Reply:
This fixes the 'free' bug:

whenever all robots get out of vision, 'weHaveRobots()' returns false, so the trees stop ticking.
this causes robotdealer to not update for the last robot removed from world and stuff breaks,
so this will tick runOneLoopCycle one more time, so robotdealer has time to remove the robots properly.

Reply:
my bad though, wasnt clear from the code itselves :)

Reply:
Wow. Impressive. How did you find that?